redis_commands.c

```diff
@@ -3911,12 +3911,15 @@ void redis_setoption_handler(INTERNAL_FUNCTION_PARAMETERS,
                 zend_string_release(redis_sock->prefix);
                 redis_sock->prefix = NULL;
             }
-            if (val_str && val_len > 0) {
-                redis_sock->prefix = zend_string_init(val_str, val_len, 0);
+            val_str = zval_get_string(val);
+            if (ZSTR_LEN(val_str) > 0) {
+                redis_sock->prefix = val_str;
+            } else {
+                zend_string_release(val_str);
             }
             RETURN_TRUE;
         case REDIS_OPT_READ_TIMEOUT:
-            redis_sock->read_timeout = atof(val_str);
+            redis_sock->read_timeout = (float)zval_get_double(val);
```

Comment:
The test for reading the timeout option is failing:
https://github.com/phpredis/phpredis/blob/bfaa89a8ae9b6a82dd8f3156e150ae898ee201be/tests/RedisTest.php#L4844-L4850

Do we need to use `(float)` here or can we stick with the double like `atof()` would normally return?

Reply:
Sorry, I thought `atof` returned a `float`. I have fixed it.